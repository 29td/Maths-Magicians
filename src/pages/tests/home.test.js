import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Home from '../home';

test('should contain element ', () => {
  const homePage = render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>,
  );
  expect(homePage).toMatchSnapshot();
});
