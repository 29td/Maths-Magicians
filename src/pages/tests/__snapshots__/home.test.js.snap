// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain element  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-container"
      >
        <nav>
          <h1
            class="logo"
          >
            Math Magician
          </h1>
          <p>
            by NR MNDEBELE
          </p>
          <ul>
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/Calculator"
              >
                Calculator
              </a>
            </li>
            <li>
              <a
                href="/pages/quotes"
              >
                Motivation
              </a>
            </li>
          </ul>
        </nav>
        <main>
          <h1>
            Welcome Math Magician!
          </h1>
          <p
            class="home-text"
          >
            Mathematics is the science and study of quality, structure, space, and change. Mathematicians seek out patterns, formulate new conjectures, and establish truth by rigorous deduction from appropriately chosen axioms and definitions.
          </p>
          <div
            class="ctc-btns"
          >
            <a
              href="/Calculator"
            >
              <button
                type="button"
              >
                Use Calculator
              </button>
            </a>
            <a
              href="/pages/quotes"
            >
              <button
                type="button"
              >
                Get Motivation
              </button>
            </a>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-container"
    >
      <nav>
        <h1
          class="logo"
        >
          Math Magician
        </h1>
        <p>
          by NR MNDEBELE
        </p>
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/Calculator"
            >
              Calculator
            </a>
          </li>
          <li>
            <a
              href="/pages/quotes"
            >
              Motivation
            </a>
          </li>
        </ul>
      </nav>
      <main>
        <h1>
          Welcome Math Magician!
        </h1>
        <p
          class="home-text"
        >
          Mathematics is the science and study of quality, structure, space, and change. Mathematicians seek out patterns, formulate new conjectures, and establish truth by rigorous deduction from appropriately chosen axioms and definitions.
        </p>
        <div
          class="ctc-btns"
        >
          <a
            href="/Calculator"
          >
            <button
              type="button"
            >
              Use Calculator
            </button>
          </a>
          <a
            href="/pages/quotes"
          >
            <button
              type="button"
            >
              Get Motivation
            </button>
          </a>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
